homeassistant/components/recollect_waste/sensor.py

```diff
@@ -128,9 +129,8 @@ def update_from_latest_data(self) -> None:
         """Update the state."""
         pickup_event = self.coordinator.data[0]
         next_pickup_event = self.coordinator.data[1]
-        next_date = str(next_pickup_event.date)
 
-        self._state = pickup_event.date
+        self._state = as_utc(pickup_event.date)
```

Comment:
You should also isoformat the dates.
```suggestion
        self._state = as_utc(pickup_event.date).isoformat()
```

Reply:
And then add the `TIMESTAMP` device class.